src/+otp/+utils/+solvers/dae34.m

```diff
@@ -0,0 +1,156 @@
+function [sol, y] = dae34(f, tspan, y0, options)
+% See
+%    Cameron, F., Palmroth, M., & Piché, R. (2002). 
+%    Quasi stage order conditions for SDIRK methods. 
+%    Applied numerical mathematics, 42(1-3), 61-75.
```

Comment:
The method in this paper is order 2 not order 3.  I think a method suitable for both index-2 and -3 DAEs would be preferable if we can find a good one.

Reply:
This is the only SDIRK (not ESDIRK) method with an embedded method that I could find with rational coefficients. Everything else is either not embedded or not rational (so single precision most of the time). If we have good coefficients I am all for it.

On order: are you sure it is order 2? So is the embedded method order 1?

Reply:
On index: I literally cannot find single method in the literature for index-3 problems. I think I have found a IRK method without an embedded method, but that's it. 

Again, if you can provide the coefficients, even if it is for IRK, we can work with it.